### ISSUE TITLE
Missing wiki info

### ISSUE THREAD

**Issue by Amazinite** [endless-sky/endless-sky#4511]
It's been noted that the wiki is missing information and has been for some time now. I'd like to use this issue as a checklist for missing information. Some missing information has already been mentioned in other issues, but I'd like to use this issue as a master list so that we can handle this issue in its entirety. If you notice anything is missing, mention it here, I'll add it to a checklist, then it will be checked off once added.

- [x] Interface information?
- [x] Creating starts (#5512)
- [x] Style guide information on Remnant and Coalition speech
- [x] #4391 and #4749
- [x] #4801 
- [x] #5413 
- [x] #5414
- [x] #5426 
- [x] #5302
- [x] #5334 
- [x] update to behavior of LocationFilters for a mission's stopover and destination, https://github.com/endless-sky/endless-sky/pull/4867
- [x] Creating phrases (especially now after #4538)
- [x] Logs (where the log book is, how logs are made, special logs)
- [x] Engine flare size guidelines (#4451)
- [x] Creating systems and planets (#3751, #1914, system attributes)
- [x] Creating minables
- [x] Landing messages and solar wind/power
- [x] Tier list update
- [x] News
- [x] How license outfits are made/work
- [x] #3128 
- [x] #4596 
- [x] #3187 
- [x] "penalty for" (as seen on the Indigenous government) and the default values
- [x] #4179
- [x] `"attitude toward"` < 0.05
- [x] #2629
- [x] #4514 (and 7863b17)
- [x] Creating fleets
- [x] `priority` and `minor` intentionally don't work with `landing` (#4752)
- [x] If multiple mission can offer, missions that come first when alphabetized will offer first. (ref: 4c31a16, #4761)
- [x] Ship leaks
- [x] #5023 (#4522, #4525, #4519, #4521, https://github.com/endless-sky/endless-sky/pull/5023/commits/4367cb69e7918b43d1400d3e240cfe810a7cf78b, #4539, #5055)

**Comment by shitwolfymakes** [comment]
In addition to logs, everything in CreatingMissions refering to the pre-0.9.9 standards needs to be removed, as it is now deprecated

**Comment by Fzzr** [comment]
- [ ] Add `licenses` documention to "Creating Outfits".
- [ ] Add `news` to feature documentation in "Creating Plugins"
- [ ] Mention or imply the existance of the Remnant, Wanderers, and Coalition in "Major Organizations". (Maybe.)

**Comment by Amazinite** [comment]
>Mention or imply the existance of the Remnant, Wanderers, and Coalition in "Major Organizations". (Maybe.)

Not even the Hai are mentioned there, so I don't consider it necessary.

**Comment by Amazinite** [comment]
>everything in CreatingMissions refering to the pre-0.9.9 standards needs to be removed, as it is now deprecated

Such as? Be specific, please.
I mentioned in the issue how people noting that things are missing from the wiki has been the case for a long time now, but generally people talk about things missing in the general sense instead of being specific, which really doesn't help all that much.
(I could read through the entirety of the wiki to figure out what's missing and what needs added, but if people are claiming to already notice missing or incorrect things, it makes more sense if they just be specific about it instead of having someone else retread the same ground.)

**Comment by tehhowch** [comment]
The News system should have a partial mention in the "Creating Events" page.

I think it does not make sense to duplicate the content of the map editor "wiki" w.r.t. explanation of the variables for systems and planets, so we should instead work on updating that repo and its documentation to be compatible with the latest game version. Then we can easily link to that documentation. It probably makes sense to create some markdown files in that repo that describe the process of creating a planet, system, galaxy, and trade network rather than a non-vc wiki.

**Comment by Bladewood** [comment]
[https://github.com/endless-sky/endless-sky/pull/2629](https://github.com/endless-sky/endless-sky/wiki/CreatingShips) I think it should be mentioned somewhere that you can swizzle individual ships.

**Comment by tehhowch** [comment]
I updated the wiki today with information about News and using #3128 . Feedback is welcome.

**Comment by tehhowch** [comment]
Finished the page for creating governments and marked 3187 + "attitude towards" + "penalty for" items as done.

**Comment by Amazinite** [comment]
Added the information for #4596 to CreatingOutfits and information for #2629 to CreatingShips.

**Comment by shitwolfymakes** [comment]
@tehhowch Where is the page for creating governments?

**Comment by shitwolfymakes** [comment]
Also, under [Triggers](https://github.com/endless-sky/endless-sky/wiki/CreatingMissions#triggers), logs are never mentioned as something the user is able to add. I only found out about it and added support for it to ESMB by looking in the actual game files

**Comment by MCOfficer** [comment]
@shitwolfymakes https://github.com/endless-sky/endless-sky/wiki/CreatingGovernments

**Comment by shitwolfymakes** [comment]
Awesome, I didn't find it because it's [here](https://github.com/endless-sky/endless-sky/wiki#creating-ships-missions-artwork-etc), not on the front page of the wiki

**Comment by tehhowch** [comment]
> Also, under [Triggers](https://github.com/endless-sky/endless-sky/wiki/CreatingMissions#triggers), logs are never mentioned as something the user is able to add. I only found out about it and added support for it to ESMB by looking in the actual game files

If you note in the OP, logs are not yet completed.

> Awesome, I didn't find it because it's not [here](https://github.com/endless-sky/endless-sky/wiki#creating-ships-missions-artwork-etc), not on the front page of the wiki

Not every page will be referenced in the shortlist. There is a "Page" expandable element that lists every page in the wiki. If you have layout or organizational suggestions, by all means, they are welcomed. (A simple sketch would be sufficient.)

**Comment by shitwolfymakes** [comment]
> If you note in the OP, logs are not yet completed.
I don't follow. Do you mean they haven't been fully implemented in the game yet?

> Not every page will be referenced in the shortlist
Given that pretty much everything else the user can create (aside from planets and systems) is under that section of the front page, I think it should definitely be there too.

**Comment by Amazinite** [comment]
>I don't follow.

Very first thing on the checklist.
![image](https://user-images.githubusercontent.com/17688683/68636736-7de0a280-04ca-11ea-8bad-013dde7f3569.png)

**Comment by shitwolfymakes** [comment]
Shit, haven't looked at that in weeks. Gotcha.

**Comment by shitwolfymakes** [comment]
We need a wiki page for CreatingFleets

**Comment by Amazinite** [comment]
For the latest issue on the list, how does this sound as an addition to Creating Missions? Everything past the first sentence ("The... unique.") is new.

> ```html
> mission <name>
> ```
> 
> The mission name must be unique. Missions are stored by the game in alphabetical order, meaning that missions will also offer in alphabetical order if multiple are able to be offered at the same time. Therefore if you have multiple missions with similar names and the same offer conditions, whichever one you want to offer first should come first alphabetically. An example of this in game is with the set of missions prefixed "FW Pug 2C", where the jump drive variant of the mission is named "FW Pug 2C: **A** Jump Drive" to force it to offer before "FW Pug 2C: **H**yperdrive" and "FW Pug 2C: **S**cram Drive".

@tehhowch @NomadicVolcano

**Comment by NomadicVolcano** [comment]
The sorting is not alphabetical.  It is done by ASCII lexical ordering, which means capital letters are before lower-case letters.  This might be easier for people to manage:

1. `FW Pug 2C**1**: Some mission that isn't really alphabetized`
2. `FW Pug 2C**2**: Yet another non-alphabetized mission`
3. `FW Pug 2C**3**: Aardvark`

Notice the number after the C.

Depending on the situation you may be able to do this:

    mission "FW Pug 2C: Aardvark"
        to offer:
            has "FW Pug 2C: Yet another non-alphabetized mission: offered"

This may work in situations where it didn't before if my pull request #4764 is merged.  It will result in reconsidering offering missions after some are completed or failed.  Essentially, it is a loop:

1. Consider one landing `to offer` job (or spaceport if we're in the spaceport)
2. Go to step 1 if any `to offer` jobs were found
3. Consider one job that should be completed or failed. (This is the step I added.)
4. Update any missions that were not able to be offered due to insufficient cargo or passenger space
5. Go to step 1

**Comment by Amazinite** [comment]
>The sorting is not alphabetical. It is done by ASCII lexical ordering, which means capital letters are before lower-case letters.

I'd say it's better to keep the wiki in layman's terms though, and saying that it's alphabetical (but perhaps mentioning ASCII lexical ordering) seems good enough to me.

**Comment by NomadicVolcano** [comment]
> > The sorting is not alphabetical. It is done by ASCII lexical ordering, which means capital letters are before lower-case letters.
> 
> I'd say it's better to keep the wiki in layman's terms though, so saying that it's alphabetical (but perhaps mentioning ASCII lexical ordering) seems good enough to me.

Forget technical feasibility for a moment and think about what the ideal solution would be.  Is there something we could put in the mission definitions to make the ordering clear and unambiguous?

Requiring the prerequisite to be offered should be enough unless I'm missing something.

**Comment by Amazinite** [comment]
>Requiring the prerequisite to be offered should be enough unless I'm missing something.

See the FW Pug 2C missions in question. Having a hyperdrive and a jump drive installed at the same time means that both "FW Pug 2C: Hyperdrive" and "FW Pug 2C: Jump Drive" could potentially offer, but since the missions have the same prefix and H comes before J, the hyperdrive version would always be offered over the jump drive version, replacing your hyperdrive with a second jump drive.

So in this case, the offer order is coming down to the mission name alone and isn't something that conditions can deal with.

**Comment by NomadicVolcano** [comment]
There is one simplification we could make.  We can implement a more understandable sorting.  It is easy to give std::set a custom comparison operator.  We could do a case-less comparison so `a` and `A` are the same.  Other measures of sanity could be added too, like considering multiple white spaces as a single space character.  It will make the mission ID comparison easier to understand.

**Comment by NomadicVolcano** [comment]
Actually, sorting sentences word-by-word and ignoring punctuation would make the most sense to people who aren't programmers.

**Comment by NomadicVolcano** [comment]
I would say a case-less comparison would be a close second, and easier for programmers to understand.

**Comment by NomadicVolcano** [comment]
I just tried three sorts:

1. lexical (current method)
2. lexical after capitalizing everything
3. ignore all characters except numbers, letters, and space; also capitalize everything

The missions currently in the file do not reorder, so the change is safe for now.

**Comment by NomadicVolcano** [comment]
Kills by an `npc` of the `"Escort"` government count towards the player's `combat rating`, but this is not mentioned in the wiki anywhere.  There should be a warning about that in the area where you are informed of NPC definitions.

**Comment by NomadicVolcano** [comment]
I put it in the discussion for that ticket.  I'll close this issue.

**Comment by Amazinite** [comment]
Well `"Escort"` *is* the government of *your fleet*, so I'd expect such behavior.

**Comment by tehhowch** [comment]
Added a checkbox for Phrases now that #4538 is merged.

**Comment by MCOfficer** [comment]
#4895 broke most links to the data files. I only checked the one [here](https://github.com/endless-sky/endless-sky/wiki/CreatingGovernments#hails), but i assume there are more.

**Comment by Zitchas** [comment]
I'm generally OK with how the [story ideas page](https://github.com/endless-sky/endless-sky/wiki/Story-Ideas) reads, however having some kind of note regarding stagnation might be appropriate. Many space games assume that more or less all space-faring races are progressing at the same rate, so the fact that some races in ES are still very active and yet almost completely stagnant is probably worth mentioning.

**Expansion, Stagnation, & Tech Development**
Every race seems to have its own technological plateau at which their development has flattened out. Where exactly this plateau is seems dependent on a variety of factors, not the least of which include how much space they have in the galaxy, what sort of threats they face on a regular basis, and their general mindset. As a general rule, most of the alien races are significantly more advanced than humanity, but they are already approaching the maximum density of colonized worlds for their area and are to some degree stagnated at their current level of technology. 

In mathematical terms, the technological level of a society appears to mimic a bell curve: slow to start, then picks up a lot of speed for the first while, then slows to a crawl again as they reach a point where they can comfortable sustain themselves and manage their threat environment. In some cases, they have even regressed somewhat. As such, while the tiers listed represent the various races' and factions' approximate technological level at the start of the game, they are not fixed at those levels and may change over the course of the stories.

**Regarding the tiers:**

I don't know how picky we want to be. I'd sort of like to have Hai listed as 1.1, since they are clearly ahead of humanity in most regards (their carrier has 40% more health than the human carrier; their cruiser equivalent has roughly 40% more health than the cruiser, kestrel, or dreadnought; their engines are generally considered to be noticeably more advanced, etc. That being said, the tier system breaks down in the decimal points, so I'm not really sure it is necessary.

In regards to the Remnant... I must admit I prefer the "Remnant ?" or "Remnant - Inconsistent" listing that they have tended to had up until now. They are rather unique in the fact that they are the only faction to have such a wide disparity of technological effectiveness; not to mention the only faction that has an affinity for understanding and adapting alien technology.

**Comment by Amazinite** [comment]
> I don't know how picky we want to be. I'd sort of like to have Hai listed as 1.1... In regards to the Remnant... I must admit I prefer the "Remnant ?" or "Remnant - Inconsistent" listing that they have tended to had up until now.

I think that all ties into the extra blurb I added about how "tiers are not a concrete 'X faction must be of Y strength in all ways because its tier is Z.'" Yes the Hai are stronger than human in some regards, but going along with the comparison I added between the Remnant and Coalition, one could instead say that that means that the Hai are a "high tier 1" while humans are a "low tier 1" instead of needing to divide the list up into a bunch of decimal values to pinpoint exactly where each faction lies when compared to the other factions.

**Comment by Zitchas** [comment]
OK, that sounds good, then. 

I do still like the idea of adding a bit about stagnation (or stabilization, as most Hai, Coalition, and Wanderers would probably term it) being a standard feature of most civilizations. Circumstances may disrupt this, but most civilizations seem to aim for stability.

**Comment by Amazinite** [comment]
Yeah, I'll add that later along with some other ideas I've had that are probably worth writing down.

**Comment by shitwolfymakes** [comment]
Bump again for a `CreatingFleets` page. I don't have wiki write access to create it myself, nor do I have the knowledge required to understand the code so I could write the wiki article well.

**Comment by Amazinite** [comment]
Done https://github.com/endless-sky/endless-sky/wiki/CreatingFleets. Will need updated following the merging of #4704, and I've already created a separate header for "Variants" on the page currently.

**Comment by Amazinite** [comment]
One of the points on the checklist is "Interface information" with a question mark. I put that question mark there because on the development roadmap, one of the items is "More generic UI code to make it easier to alter the user interface." Given that the intention is to change the UI code down the road, should we wait until that's finished to put that information on the wiki? Or should we put the current information on the wiki (even if it's a little useless given the relative inflexibility).

**Comment by tehhowch** [comment]
Added a checkbox for https://github.com/endless-sky/endless-sky/pull/4867, will get to this tomorrow

> I've finished basically everything on the checklist outside of two things.
> 
> One of the points on the checklist is "Interface information" with a question mark. I put that question mark there because on the development roadmap, one of the items is "More generic UI code to make it easier to alter the user interface." Given that the intention is to change the UI code down the road, should we wait until that's finished to put that information on the wiki? Or should we put the current information on the wiki (even if it's a little useless given the relative inflexibility).

I'd need to research a bit to determine when that was added, relative to the refactorings of interface code that have already occurred. There are a lot more changes that _could_ be made to the interface code, especially with regards to rendering groups of a given item in particular locations. I think it is fine to document what we have now, as this syntax won't be disappearing anytime soon.

**Comment by tehhowch** [comment]
Updated the CreatingMissions page with the new filter changes, and made it more clear that filters work for waypoint and stopover definitions, too.

**Comment by tehhowch** [comment]
Looks like "More generic UI code to make it easier to alter the user interface" was added to the roadmap on [jan 2, 2016](https://github.com/endless-sky/endless-sky/wiki/DevelopmentRoadmap/37234f3820c6d9b2dd06de7fc19d0eb4fe4cc361)

At that point, the Interface class was considerably less generic, and was missing some current box and anchoring functionality.
The biggest refactor since then was probably 1a7dd1e33e8070dd4cbf6089b82575f71e239757.

That said, having the interface class support specifying things a given way doesn't mean the game code drawing a panel even cares about what the interface says. Lots of refactoring is needed to get rid of the plethora of imperative point / line-by-line layout instructions within the `___Panel`s.

**Comment by Amazinite** [comment]
I've checked everyone on the list except for the interface information. (And in working on finishing everything else, I actually found out that we were missing information on launch effects, which I've now added to CreatingShips.) Going forward, I'd expect us to update the wiki just before or after merging pull requests that add to or change the data syntax, which I have been doing with many of my recent merges.

**Comment by shitwolfymakes** [comment]
In [CreatingMissions](https://github.com/endless-sky/endless-sky/wiki/CreatingMissions), the example for the `distance` attribute of a location filter is only properly documented as `distance [<min#>] <max#>` in the paragraph that describes what it does. Everywhere else it is incorrectly described as `distance [[<min>] <max>]`

**Comment by Amazinite** [comment]
Fixed.

**Comment by EricFromCanada** [comment]
In [BuildInstructions](https://github.com/endless-sky/endless-sky/wiki/BuildInstructions#mac-os-x), with #6121 having been merged, the following changes can be made:
- change both the heading and the page's first sentence to use "macOS" instead of "Mac OS X"
  - and also in [HelpWanted](https://github.com/endless-sky/endless-sky/wiki/HelpWanted) & [CreatingPlugins](https://github.com/endless-sky/endless-sky/wiki/CreatingPlugins)
- remove `brew install sdl2` (since Xcode now downloads SDL2 on its own) and combine remaining commands into `brew install libmad libpng jpeg-turbo`
- remove the entire "Library Paths" section, since this is now handled by a [shell script](https://github.com/endless-sky/endless-sky/blob/master/utils/set_dylibs_rpath.sh) within Xcode
- in the preceding paragraph about modifying search paths, change "to point to wherever Homebrew installed those libraries" to "in case your Homebrew for Intel Mac installation does not use the standard `/usr/local` prefix".

**Comment by shitwolfymakes** [comment]
The example mission grammar still lists `illegal` and `stealth` nodes as children of `cargo`. This was removed in v0.9.9 and onward

**Comment by shitwolfymakes** [comment]
`dialog phrase <text>` documentation needs to be added under [Triggers](https://github.com/endless-sky/endless-sky/wiki/CreatingMissions#triggers)

Also, documentation for all parts of `dialog` to reflect how inline phrase declaration works

**Comment by Anarchist2** [comment]
The interaction between stopovers and things that use jump distance (deadlines, payments) isn't documented anywhere.

**Comment by Amazinite** [comment]
Added the following line to both the deadline and payment information in CreatingMissions:
> If the mission has waypoints or stopovers then the number of jumps needed to visit all waypoints and stopovers before reaching the destination is used.

**Comment by shitwolfymakes** [comment]
How does it calculate that? Distance to each waypoint from origin? Distance between each waypoint in the list?

**Comment by Amazinite** [comment]
It does an approximation of a traveling salesman path.
